pkg/ui/handlers.go

```diff
@@ -1,37 +1,55 @@
 package ui
 
 import (
+	"fmt"
 	"io/fs"
 	"net/http"
-	"path"
+	"regexp"
+	"strings"
 
 	"github.com/canonical/identity-platform-login-ui/internal/logging"
 	"github.com/go-chi/chi/v5"
 )
 
 type API struct {
-	fileServer http.Handler
-
-	logger logging.LoggerInterface
+	fileServer   http.Handler
+	fileSystem   fs.FS
+	servePattern *regexp.Regexp
+	logger       logging.LoggerInterface
 }
 
 func (a *API) RegisterEndpoints(mux *chi.Mux) {
 	// TODO @shipperizer unsure if we deal with any POST/PUT/PATCH via js
-	mux.HandleFunc("/*", a.uiFiles)
+	mux.HandleFunc("/ui/*", a.uiFiles)
+
 }
 
 // TODO: Validate response when server error handling is implemented
 func (a *API) uiFiles(w http.ResponseWriter, r *http.Request) {
-	if ext := path.Ext(r.URL.Path); ext == "" && r.URL.Path != "/" {
-		r.URL.Path += ".html"
+
+	if match := a.servePattern.FindString(r.URL.Path); match == ".html" {
+		w.WriteHeader(http.StatusNotFound)
+		// To make the output consistent with the fileserver
+		w.Write([]byte(fmt.Sprintf("%v page not found\n", http.StatusNotFound)))
+		return
+	}
+
+	if match := a.servePattern.FindString(r.URL.Path); match == "" {
+		r.URL.Path = fmt.Sprintf("%s.html", r.URL.Path)
 	}
 
+	r.URL.Path = strings.TrimPrefix(r.URL.Path, "/ui")
```

Comment:
`/ui` is used in 2 parts of the code, let's move it to a `const`.

Reply:
done